src/coreComponents/physicsSolvers/fluidFlow/DissipationCompositionalMultiphaseFVMKernels.hpp

```diff
@@ -0,0 +1,469 @@
+   * @param[in] iconn the connection index
+   * @param[inout] stack the stack variables
+   */
+  GEOS_HOST_DEVICE
+  void computeFlux( localIndex const iconn,
+                    StackVariables & stack ) const
+  {
+
+    m_stencilWrapper.computeWeights( iconn,
+                                     m_permeability,
+                                     m_dPerm_dPres,
+                                     stack.transmissibility,
+                                     stack.dTrans_dPres );
+
+    // ***********************************************
+    // First, we call the base computeFlux to compute:
+    //  1) compFlux and its derivatives,
+    //
+    // We use the lambda below (called **inside** the phase loop of the base computeFlux) to compute dissipation terms
+    Base::computeFlux( iconn, stack, [&] ( integer const ip,
+                                           localIndex const (&k)[2],
+                                           localIndex const (&seri)[2],
+                                           localIndex const (&sesri)[2],
+                                           localIndex const (&sei)[2],
+                                           localIndex const connectionIndex,
+                                           localIndex const k_up,
+                                           localIndex const er_up,
+                                           localIndex const esr_up,
+                                           localIndex const ei_up,
+                                           real64 const & potGrad,
+                                           real64 const & phaseFlux,
+                                           real64 const (&dPhaseFlux_dP)[2],
+                                           real64 const (&dPhaseFlux_dC)[2][numComp] )
+    {
+      GEOS_UNUSED_VAR( k_up, potGrad, phaseFlux, dPhaseFlux_dP, dPhaseFlux_dC, er_up, esr_up, ei_up );
+
+      /// Storing dissipation flux and its derivatives locally
+      real64 dissFlux[numComp]{};
+      real64 dDissFlux_dP[numFluxSupportPoints][numComp]{};
+      real64 dDissFlux_dC[numFluxSupportPoints][numComp][numComp]{};
+      real64 fluxPointCoef[numFluxSupportPoints] = {1.0, -1.0}; // for gradients
+
+      real64 viscosityMult[3] = {1.0, 1.0, 1.0}; // for viscosity
+
+      /// Step 1. Calculate the continuation parameter based on the current Newton iteration
+      real64 kappaDBC = 1.0; // default value
+
+      if( m_continuationDBC ) // if continuation is enabled
+      {
+        if( m_curNewton >= 5 )
+        {
+          kappaDBC = m_kappaminDBC;
+        }
+        else
+        {
+          for( int mp = 0; mp < m_curNewton; mp++ ) kappaDBC *= m_contMultiplierDBC;
+          kappaDBC = std::max( kappaDBC, m_kappaminDBC );
+        }
+      }
+      /// Step 2. Collect all contributions
+      real64 poreVolume_n = 0; // Pore volume contribution
+      real64 trans = 0; // Transmissibility contribution
+      for( integer ke = 0; ke < numFluxSupportPoints; ++ke )
```

Comment:
IIUC we are assuming TPFA, hence the `0.5` prefactor. It is not as such a strong assumption but can we comment about it just for maintenance purpose

Reply:
Done.